HIRS_Utils/src/test/java/hirs/ima/matching/ImaAcceptableHashRecordMatcherTest.java

```diff
@@ -0,0 +1,153 @@
+        IMABaselineRecord baselineRecord = new IMABaselineRecord(FILENAME_1, HASH_1);
+        Assert.assertEquals(
+                new ImaAcceptableHashRecordMatcher(
+                        Collections.singletonList(baselineRecord),
+                        SimpleImaBaselineTest.getTestImaPolicy(false),
+                        baseline).contains(measurementRecord),
+                new IMAMatchStatus<>(
+                        measurementRecord, ReportMatchStatus.MATCH, baselineRecord, baseline
+                )
+        );
+    }
+
+    /**
+     * Tests that contains returns the expected 'UNKNOWN' match status if no record matches
+     * the collected measurement in a nonempty baseline.
+     */
+    @Test
+    public void testContainsNonEmptyButUnknown() {
+        SimpleImaBaseline baseline = getTestSimpleImaBaseline();
+        IMAMeasurementRecord measurementRecord = new IMAMeasurementRecord(FILENAME_1, HASH_1);
+        IMABaselineRecord baselineRecord = new IMABaselineRecord(FILENAME_1, HASH_2);
+        Assert.assertEquals(
+                new ImaAcceptableHashRecordMatcher(
+                        Collections.singletonList(baselineRecord),
+                        SimpleImaBaselineTest.getTestImaPolicy(false),
+                        baseline).contains(measurementRecord),
+                new IMAMatchStatus(measurementRecord, ReportMatchStatus.UNKNOWN, baseline)
+        );
+    }
+
+    /**
+     * Tests that contains returns all matching baseline records from a baseline when there are
+     * multiple matches to a given measurement record.
+     */
+    @Test
+    public void testContainsMultipleMatchingBaselineRecords() {
+        SimpleImaBaseline baseline = getTestSimpleImaBaseline();
+        IMAMeasurementRecord measurementRecord = new IMAMeasurementRecord(FILENAME_1, HASH_1);
+        Set<IMABaselineRecord> baselineRecords = new HashSet<>(Arrays.asList(
+                new IMABaselineRecord(FILENAME_1, HASH_1),
+                new IMABaselineRecord(FILENAME_2, HASH_1),
+                new IMABaselineRecord(FILENAME_1, HASH_2)
+        ));
+        Assert.assertEquals(
+                new ImaAcceptableHashRecordMatcher(
+                        baselineRecords,
+                        SimpleImaBaselineTest.getTestImaPolicy(false),
+                        baseline
+                ).contains(measurementRecord),
+                new IMAMatchStatus<>(
+                        measurementRecord,
+                        ReportMatchStatus.MATCH,
+                        new HashSet<>(Arrays.asList(
+                            new IMABaselineRecord(FILENAME_1, HASH_1),
+                            new IMABaselineRecord(FILENAME_2, HASH_1))
+                        ),
+                        baseline
+                )
+        );
+    }
+
+    /**
+     * Tests that contains functions if a matching record is given in the case of that matching
```

Comment:
This javadoc does not describe what the test is doing.

Reply:
Thanks, will fix.

Reply:
looks good.